### ISSUE TITLE
Attack Speed Buffs

### ISSUE THREAD

**Issue by lokka30** [ArcanePlugins/LevelledMobs#432]
At some point in LM 4.x, we can add an attack speed buff system. Allowing levelled mobs to have a chance to make surprise attacks, especially on ranged mobs, would take this plugin to the next level in making gameplay interesting. As Oath put it, "Basically we've got a one-punch-man generator and not killing machine generator. One skeleton shooting the same insta-kill arrow every 3 seconds, or them suddenly launching three in the span of a second"

<details>
<summary>Oath's Implementation Idea</summary>

Adding a multi-attack setting:
I know that we can't actually use the attribute for attack speed as that's player only, however I did discover this little gem:
https://hub.spigotmc.org/javadocs/spigot/org/bukkit/entity/LivingEntity.html#swingMainHand()
TLDR ability to active a swing on the main or secondary hand if entity has arms to swing. Was imagining this as a form of attack speed but applied as a multi-attack. When an entity attacks with their vanilla base attack, in the space between that and the next vanilla attack, these multi attacks could occur. Time between TBD but concept (using LM3 syntax):

```yaml
apply-settings:
  multi-attack:
    primary-hand:  # Regarding 'swingMainHand'
      activate: 1-2  # Number of additional swings as a randomly selected value between range
      chance: 0.1 # Percent chance of occurring after each vanilla 'parent' attack
    secondary-hand: # Regarding 'swingOffhand'
      activate: 1
      chance: 0.1
    combine-attacks: true # If false, if chance succeeded on 'primary-hand' or 'secondary-hand', then the other will not be processed. If true, then both could potentially occur (combo critical attacks)
```

```yaml
  apply-settings:
    critical-hit-chance: 0.15 # Percent chance to occur with every landed hit
    critical-hit-modifier: 0.20 # Percent increase to the damage occurred
```

</details>

Also an idea of a shield mechanic which might be worth making into a different issue sometime

> Might be worth investigating; a 'shield' multiplier where entities can Y percent randomly shield themselves from up to X damage, where X is a percentage of their current health (so the lower the health, the weaker the entity, the less likely they'll be able to defend; flip side being stronger the entity the more likely they'll shield).
